### PR TITLE
Better contact related errors

### PR DESCRIPTION
**Pull request by rowanseymour**
* ContactGroup.update_contacts now prevents adding of blocked contacts to groups
* Contact bulk actions endpoint errors if any passed UUIDs are invalid (or test contacts)
* Contact write endpoint validates that a blocked contact isn't being added to a group
* Added REST framework exception handler so that API calls don't return the html error page
* Simplified `Contact.release()`

### CODE REVIEW

File: temba/settings_common.py [hunk 1]
```diff
@@ -985,6 +985,9 @@ def __getitem__(self, item):
     )
 }
 
+if not TESTING:
```
Comment (rowanseymour) [inline temba/settings_common.py]: During testing we don't want this swallowing exceptions

File: temba/contacts/models.py [hunk 2]
```diff
@@ -872,29 +872,26 @@ def release(self):
         """
         Releases (i.e. deletes) this contact, provided it is currently not deleted
         """
-        # perform everything in an org level lock to prevent conflicts with get_or_create or update_urns
-        with self.org.lock_on(OrgLock.contacts):
```
Comment (rowanseymour) [inline temba/contacts/models.py]: Don't need lock here now because `update_urns` uses lock

**Comment by rowanseymour** [comment]
Talked myself out of having `update_groups` implicitly unblock contacts as it seems a little weird in the API case. So instead API endpoints return an error if trying to add a blocked contact to a group, and `Contact.create_instance` calls `unblock` to cover the import case.

**Comment by nicpottier** [comment]
looks sane

**Pull request merged**